{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aee7f747",
   "metadata": {},
   "source": [
    "## Load and explore the DREAM7 datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f3b44051",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data shapes:\n",
      "Full drug response data shape: (53, 28)\n",
      "Training drug response data shape: (35, 28)\n",
      "Test drug response data shape: (18, 28)\n",
      "Classification drug response data shape: (53, 28)\n",
      "Data shapes:\n",
      "RNA-seq quantification data shape: (36953, 45)\n",
      "RNA-seq expressed calls data shape: (36953, 45)\n",
      "mRNA gene expression data shape: (18632, 46)\n",
      "Copy number variation data shape: (27234, 48)\n",
      "Methylation data shape: (27551, 44)\n",
      "Exome sequencing data shape: (33160, 26)\n",
      "RPPA protein expression data shape: (131, 43)\n",
      "MSigDB hallmarks annotation data shape: (42995, 50)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "\n",
    "# DRUG RESPONSE DATA in the form of -1log10(GI50) values. The GI50 is the concentration of drug needed to inhibit growth of the cell line by 50%.\n",
    "# taking the -1log10 makes higher values better response (lower concentration needed). a value of 6 means 1 micromolar concentration needed, 5 means 10 micromolar, etc.\n",
    "drug_response_full = pd.read_csv('drug_response_full_matrix.csv', index_col=0)\n",
    "drug_response_train = pd.read_csv('drug_response_train_matrix.csv', index_col=0)\n",
    "drug_response_test = pd.read_csv('drug_response_test_matrix.csv', index_col=0)\n",
    "drug_response_classification = pd.read_csv('drug_response_classification.txt', sep='\\t', index_col=0)\n",
    "print(\"Data shapes:\")\n",
    "print('Full drug response data shape:', drug_response_full.shape)\n",
    "print('Training drug response data shape:', drug_response_train.shape)\n",
    "print('Test drug response data shape:', drug_response_test.shape)\n",
    "print('Classification drug response data shape:', drug_response_classification.shape)\n",
    "\n",
    "\n",
    "# Load the molecular data, all molecules on the gene level will have gene names as rows and cell lines as columns\n",
    "\n",
    "# GENE EXPRESSION and GENE-level data\n",
    "\n",
    "# There's gene expression data in the form of RANA-seq quantification and microarrays\n",
    "RNA_seq_quant = pd.read_csv('RNAseq_quantification.txt', sep='\\t', index_col=0)\n",
    "RNA_seq_calls = pd.read_csv('RNAseq_expressed_calls.txt', sep='\\t', index_col=0)\n",
    "mRNA = pd.read_csv('GeneExpression.txt' , sep='\\t', index_col=0)\n",
    "\n",
    "# There is gene-level copy number variation data\n",
    "copy_number = pd.read_csv('SNP6_gene_level.txt', sep='\\t', index_col=0)\n",
    "\n",
    "# Methylation data exists too\n",
    "methylation = pd.read_csv('Methylation.txt', sep='\\t', index_col=0)\n",
    "\n",
    "# There's also information about mutations in genes, in exome sequencing data \n",
    "exome_seq = pd.read_csv('ExomeSeq.txt', sep='\\t', index_col=0)\n",
    "\n",
    "# PROTEIN-LEVEL DATA\n",
    "# There is protein expression data from Reverse Phase Protein Arrays (RPPA)\n",
    "rppa = pd.read_csv('RPPA.txt', sep='\\t', index_col=0)\n",
    "print(\"Data shapes:\")\n",
    "print('RNA-seq quantification data shape:', RNA_seq_quant.shape)\n",
    "print('RNA-seq expressed calls data shape:', RNA_seq_calls.shape)\n",
    "print('mRNA gene expression data shape:', mRNA.shape)\n",
    "print('Copy number variation data shape:', copy_number.shape)\n",
    "print('Methylation data shape:', methylation.shape) \n",
    "print('Exome sequencing data shape:', exome_seq.shape)  \n",
    "print('RPPA protein expression data shape:', rppa.shape)\n",
    "\n",
    "# annotation data - this was made in 2025 by annotating the MSigDB hallmarks gene sets to the genes in our data\n",
    "msig_hallmarks = pd.read_csv('genes_msig_hallmarks_annotation.csv', index_col=0)\n",
    "print('MSigDB hallmarks annotation data shape:', msig_hallmarks.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "622b6167",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "CoDIAC-dev",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
